src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -540,7 +545,8 @@ public T next() {
                     throw new NoSuchElementException();
                 }
                 // each "entry" has 3 ints in table entries
-                return (T)getZipEntry(null, res.zsrc.getEntryPos(i++ * 3));
+                int pos = res.zsrc.getEntryPos(i++ * 3);
+                return (T)getZipEntry(new EntryPos(getEntryName(pos), pos));
```

Comment:
Do we have benchmarks covering the various types of iteration?

Reply:
I realized it's better to keep the signature of `getZipEntry` so these iteration callers won't need to construct the EntryPos. I assume calling `getEntryName` should have similar performance characteristic to the existing code which calls `getEntryPos` with a null name, since we're just seem to be moving the String decoding from one method to another.

But no, I don't think we have benchmarks for ZipEntry enumeration.